runtime/address.rs

```diff
@@ -6,53 +6,56 @@ use crate::handler;
 use core::{
     fmt,
     mem::{size_of, MaybeUninit},
-    ptr, slice,
+    ptr, slice, marker::PhantomData,
 };
 use kerla_utils::alignment::align_down;
 
 /// Represents a physical memory address.
 #[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
 #[repr(transparent)]
-pub struct PAddr(usize);
+pub struct PAddr<'memory> {
```

Comment:
Use `'a` to follow a naming convention in Rust.

```suggestion
pub struct PAddr<'a> {
```

Reply:
@nuta there's no such convention - lifetimes should be named in meaningful way.